design/volume-snapshot-data-movement/volume-snapshot-data-movement.md

```diff
@@ -637,9 +645,40 @@ In DUCR/DDCR’s status, we have fields like ```totalBytes``` and ```doneBytes``
 - Call ```kubectl get dataupload -n velero xxx or kubectl get datadownload -n velero xxx```.
 - Call ```velero backup describe –details```. This is implemented as part of BIA/RIA V2, the above values are transferred to async operation and this command retrieve them from the async operation instead of DUCR/DDCR. See [general progress monitoring design][2] for details
 
+## Retain Native Snapshots
+Users are allowed to specify a global number as the number of native snapshots to be retained for each volume, then:
+- If the number is not specified, or if it is not a positive value, it means no native snapshot is to be retained
+- Otherwise, the value defines the max snapshots to be retained for each volume, a.k.a, the limit. If the limit is exceeded, the oldest retained snapshot will be removed
+
+The limit number is set as an annotation of a backup CR, the annotation is called ```backup.velero.io/data-mover-snapshot-to-retain```. In this way, CMPs are able to query the number from the backup CR.  
```

Comment:
Can you clarify what CMP stands for?

Reply:
Is `backup.velero.io/data-mover-snapshot-to-retain` for querying purposes only and populated by velero? or can user set this value at backup CR creation to modify the behavior per backup?

Should we allow per pvc annotation settings?

Reply:
Corrected it, it should be `DMP`, thanks for the reminding.

Reply:
The annotation is for querying purposes only, we don't expect users to config it, they should config the server parameter instead.
Actually, users don't have opportunity to config it, as it is added when creating the backup CR and consumed immediately.

Reply:
We don't see any usability to configure the number of retained snapshot per PVC. So it is not supported.